### ISSUE TITLE
CSP issues with Gitter sidecar

### ISSUE THREAD

**Issue by lynnbennett** [FreeCodeCamp/FreeCodeCamp#11163]
Challenge [Change Text Inside an Element Using jQuery](https://www.freecodecamp.com/challenges/change-text-inside-an-element-using-jquery#?solution=%0Afccss%0A%20%20%24%28document%29.ready%28function%28%29%20%7B%0A%20%20%20%20%24%28%22%23target1%22%29.css%28%22color%22%2C%20%22red%22%29%3B%0A%20%20%20%20%24%28%22target4%22%29.html%28%22%3Cem%3E%20jQuery%20Playground%3C%2Fem%3E%22%29%20%0A%20%20%7D%29%3B%0Afcces%0A%0A%3C!--%20Only%20change%20code%20above%20this%20line.%20--%3E%0A%0A%3Cdiv%20class%3D%22container-fluid%22%3E%0A%20%20%3Ch3%20class%3D%22text-primary%20text-center%22%3EjQuery%20Playground%3C%2Fh3%3E%0A%20%20%3Cdiv%20class%3D%22row%22%3E%0A%20%20%20%20%3Cdiv%20class%3D%22col-xs-6%22%3E%0A%20%20%20%20%20%20%3Ch4%3E%23left-well%3C%2Fh4%3E%0A%20%20%20%20%20%20%3Cdiv%20class%3D%22well%22%20id%3D%22left-well%22%3E%0A%20%20%20%20%20%20%20%20%3Cbutton%20class%3D%22btn%20btn-default%20target%22%20id%3D%22target1%22%3E%23target1%3C%2Fbutton%3E%0A%20%20%20%20%20%20%20%20%3Cbutton%20class%3D%22btn%20btn-default%20target%22%20id%3D%22target2%22%3E%23target2%3C%2Fbutton%3E%0A%20%20%20%20%20%20%20%20%3Cbutton%20class%3D%22btn%20btn-default%20target%22%20id%3D%22target3%22%3E%23target3%3C%2Fbutton%3E%0A%20%20%20%20%20%20%3C%2Fdiv%3E%0A%20%20%20%20%3C%2Fdiv%3E%0A%20%20%20%20%3Cdiv%20class%3D%22col-xs-6%22%3E%0A%20%20%20%20%20%20%3Ch4%3E%23right-well%3C%2Fh4%3E%0A%20%20%20%20%20%20%3Cdiv%20class%3D%22well%22%20id%3D%22right-well%22%3E%0A%20%20%20%20%20%20%20%20%3Cbutton%20class%3D%22btn%20btn-default%20target%22%20id%3D%22target4%22%3E%3Cem%3E%23target4%3C%2Fem%3E%3C%2Fbutton%3E%0A%20%20%20%20%20%20%20%20%3Cbutton%20class%3D%22btn%20btn-default%20target%22%20id%3D%22target5%22%3E%23target5%3C%2Fbutton%3E%0A%20%20%20%20%20%20%20%20%3Cbutton%20class%3D%22btn%20btn-default%20target%22%20id%3D%22target6%22%3E%23target6%3C%2Fbutton%3E%0A%20%20%20%20%20%20%3C%2Fdiv%3E%0A%20%20%20%20%3C%2Fdiv%3E%0A%20%20%3C%2Fdiv%3E%0A%3C%2Fdiv%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (Macintosh; Intel Mac OS X 10_11_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/53.0.2785.143 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:

``` html

<script>
  $(document).ready(function() {
    $("#target1").css("color", "red");
    $("target4").html("<em> jQuery Playground</em>") 
  });
</script>

<!-- Only change code above this line. -->

<div class="container-fluid">
  <h3 class="text-primary text-center">jQuery Playground</h3>
  <div class="row">
    <div class="col-xs-6">
      <h4>#left-well</h4>
      <div class="well" id="left-well">
        <button class="btn btn-default target" id="target1">#target1</button>
        <button class="btn btn-default target" id="target2">#target2</button>
        <button class="btn btn-default target" id="target3">#target3</button>
      </div>
    </div>
    <div class="col-xs-6">
      <h4>#right-well</h4>
      <div class="well" id="right-well">
        <button class="btn btn-default target" id="target4"><em>#target4</em></button>
        <button class="btn btn-default target" id="target5">#target5</button>
        <button class="btn btn-default target" id="target6">#target6</button>
      </div>
    </div>
  </div>
</div>

```

**Comment by lynnbennett** [comment]
i am unable to join chat and having problem solution

**Comment by erictleung** [comment]
@lynnbennett you need to look at the current text within the `target4` element and change it so it is emphasized. Right now, you're adding `<em> jQuery Playground</em>` instead of what is preexisting.

Can you explain on what you are experiencing when trying to join the chat?

**Comment by lynnbennett** [comment]
@erictleung Why did you add the blocked label?

**Comment by erictleung** [comment]
@lynnbennett we put that label there whenever we're waiting for a response from either the original poster or someone on our team. Sorry if it alarmed you in any way.

**Comment by Dayfiri** [comment]
I have the same problem with chat if I try to sign in with github for discussion. But this only happens when I'm on FreeCodeCamp page, but when I open chat in a new window, it works alright.

**Comment by erictleung** [comment]
@Dayfiri ah so you're having issues while you're using the Gitter side chat within the FreeCodeCamp website? I haven't had any issues with it. Can you share screenshots of what kind of error or issue you come across when you open the chat within the FreeCodeCamp page? Thanks!

@lynnbennett are you experiencing a similar issue with the FreeCodeCamp page or are having issues on the www.gitter.com website?

**Comment by Dayfiri** [comment]
This is the *first* image where chat is open and going. **Note** I'm signed on FCC page but not in gitter chat.
![first-image](https://cloud.githubusercontent.com/assets/16082563/19327385/0a8a61c6-90cf-11e6-8744-387a6180cf27.png)

This is *second* image where I'm trying to "start chatting" with my github account
![second-image](https://cloud.githubusercontent.com/assets/16082563/19327410/2ed083f8-90cf-11e6-95f4-bf791d7ad7a9.png)

And this is the *third* image where that shows what happens after I click to join chat with my github account. It looks like 1. image ? Well it is not because after applying for chat log in, nothing happens and I'm on starting page.
![third-image](https://cloud.githubusercontent.com/assets/16082563/19327426/4c652162-90cf-11e6-8190-90660a7de36e.png)

**Comment by erictleung** [comment]
@Dayfiri thanks so much for those very informative screenshots. That is very odd. I signed out of Gitter while being signed into FreeCodeCamp to reproduce your first image. I got to where you sign in with GitHub. I successfully signed in with GitHub and was able to get to chat again. I'm assuming that @lynnbennett is going through the same issue of unable to sign into the side bar chat.

tl;dr I was unable to reproduce your entire experience.

@Dayfiri what happens if you sign out of Gitter, sign into Gitter through another window you've created, and then return back to FreeCodeCamp? Are you still signed in? Or signed back out?

**Comment by Dayfiri** [comment]
@erictleung After login in on gitter page in a new window, I went back to Free Code Camp page. I signed out and signed in. Chat still ain't working, but this time I logged in with twitter account and same problem occurs.

**Comment by lynnbennett** [comment]
![screen shot 2016-10-13 at 12 30 40 pm](https://cloud.githubusercontent.com/assets/16314135/19357747/ee4b1d30-9140-11e6-927d-e6b0530a9c0a.png)

**Comment by raisedadead** [comment]
@Dayfiri twitter login + gitter is not supported on FreeCodeCamp chat rooms.

Please try this:
1. Logout from Github.
2. Logout from FreeCodeCamp.
3. **Clear your browser cache, cookies.**
4. Retry.

Let us know if the problem persists.

**Comment by Dayfiri** [comment]
@raisedadead I cleared everything in Firefox dev version 51 that I use. Chat only works if I open it in a new page and then log in. If I try to login from FreeCodeCamp page it always returns me to the beginning. I tried the same thing in *Opera* browser and it works alright.

**Comment by Mongues** [comment]
Hello world

**Comment by Mongues** [comment]
Okay thanks it needed checking for meeting up to date with the new

**Comment by raisedadead** [comment]
Hey @Dayfiri sorry about the issue you are facing, unfortunately, firefox dev is a non-standard browser, where changes are pushed to it on a nightly basis if I am not wrong.  And this looks like a very weird case.

I agree this is an inconvenience, but the thing is we do not do any additional configuration for the authentication for the login into Gitter. It's a sidecar page that's entirely provided by Gitter and is simply integrated into Free Code Camp, as an iFrame.

You may try opening the developer console for your browser and then, look at any errors especially CSP related logs popping up there.

**Comment by Dayfiri** [comment]
@raisedadead 
I checked console log when trying to log in into gitter chat while on Free Code Camp web page.
This is the error displayed:
```
downloadable font: Layout: tags aren't arranged alphabetically. (font-family: "fontello" style:normal weight:normal stretch:normal src index:1) source: https://cdn02.gitter.im/_s/e70b2ef/fonts/fontello/font/fontello.woff2?93021818
```

**Comment by raisedadead** [comment]
@Dayfiri thanks for the logs, it's definitely not the fonts. maybe the first warning is worth looking into.

**Comment by roddrec23** [comment]
nice

**Comment by raisedadead** [comment]
Closing as decayed. Please comment if you are still facing this?
